lib/client/session.go

```diff
@@ -75,6 +77,22 @@ type NodeSession struct {
 	enableEscapeSequences bool
 
 	terminal *terminal.Terminal
+
+	// x11Display is the client's local display.
+	x11Display string
+	// x11Proto is the xauth protocol for this session.
+	x11Proto string
+	// x11RealCookie in an xauth cookie which provides
+	// access to the client's local display.
+	x11RealCookie string
+	// x11FakeCookie is a spoof of the real cookie. This is used
+	// to authenticate x11 channel requests from the server without
+	// exposing the real cookie. The fake cookie is replaced with
```

Comment:
There isn't an RFD because initially the protocol seemed straightforward enough to implement. After finding next to no documentation and delving into the OpenSSH source code, I found a lot more complexity. I am planning to add a lot more details to the PR description about authentication, security, etc. If you think it'd be better I can write an RFD about the implementation details instead.